pkg/project/resource_project_role.go

```diff
@@ -190,33 +170,36 @@ var updateRoles = func(projectKey string, terraformRoles []Role, m interface{})
 	}
 	log.Printf("[TRACE] projectRoles: %+v\n", projectRoles)
 
-	rolesToBeAdded := difference(rolesToEquatables(terraformRoles), rolesToEquatables(projectRoles))
+	terraformRolesSet := SetFromSlice(terraformRoles)
+	projectRolesSet := SetFromSlice(projectRoles)
+
+	rolesToBeAdded := terraformRolesSet.Difference(projectRolesSet)
 	log.Printf("[TRACE] rolesToBeAdded: %+v\n", rolesToBeAdded)
 
-	rolesToBeUpdated := intersection(rolesToEquatables(terraformRoles), rolesToEquatables(projectRoles))
+	rolesToBeUpdated := terraformRolesSet.Intersection(projectRolesSet)
 	log.Printf("[TRACE] rolesToBeUpdated: %+v\n", rolesToBeUpdated)
 
-	rolesToBeDeleted := difference(rolesToEquatables(projectRoles), rolesToEquatables(terraformRoles))
+	rolesToBeDeleted := projectRolesSet.Difference(terraformRolesSet)
```

Comment:
Not sure how you implemented this, but pretty clever. Or, is it part of 1.18

Reply:
It's partly 1.18. But not as much as I'd like it can be.

Reply:
I created our own `Set` https://github.com/jfrog/terraform-provider-project/pull/36/files#diff-f0b31058c422450c709fefe032c3745ab2ae2d4f1a8719cfffdaeb6ead6bfc9f